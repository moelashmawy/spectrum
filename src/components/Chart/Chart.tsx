import React from "react";
import {
  Chart,
  ChartWrapperOptions,
  GoogleChartWrapperChartType,
} from "react-google-charts";
import styled from "styled-components";

const StyledGauge = styled.div`
  height: "100%";
  width: "100%";
  text-align: center;
`;

interface GaugeChartProps {
  type: GoogleChartWrapperChartType;
  chartData: Array<Array<string | number>>;
  chartOptions: ChartWrapperOptions["options"];
  width?: number;
  height?: number;
}

const GaugeChart: React.FC<GaugeChartProps> = ({
  type,
  chartData,
  chartOptions,
  width,
  height,
}) => {
  return (
    <StyledGauge>
      <Chart
        chartType={type}
        height={`${height}px`}
        width={`${width}px`}
        data={chartData}
        options={chartOptions}
      />
    </StyledGauge>
  );
};

export default GaugeChart;
