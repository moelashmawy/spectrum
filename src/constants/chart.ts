export const TEMP_OPTIONS = {
  redFrom: 80,
  redTo: 100,
  yellowFrom: 65,
  yellowTo: 80,
  minorTicks: 10,
  min: -100,
};

export const VELOCITY_OPTIONS = {
  redFrom: 180,
  redTo: 200,
  yellowFrom: 130,
  yellowTo: 180,
  minorTicks: 10,
  min: 0,
  max: 200,
};

export const ALTITUDE_OPTIONS = {
  vAxis: { title: "Height" },
  hAxis: { title: "Time" },
};
