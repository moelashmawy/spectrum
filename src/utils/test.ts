import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { ReactElement } from "react";

export const createSnapshot = (ComponentToTest: ReactElement) => {
  const {
    container: { firstChild: actual },
  } = render(ComponentToTest);
  expect(actual).toMatchSnapshot();
};
