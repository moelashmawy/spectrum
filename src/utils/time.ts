export const getCurrentTime = () => {
  const d = new Date();
  return d.toLocaleTimeString();
};
