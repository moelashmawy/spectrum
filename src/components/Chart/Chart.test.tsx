import Chart from "./Chart";
import { createSnapshot } from "../../utils/test";
import { VELOCITY_OPTIONS } from "../../constants/chart";

describe("Chart", () => {
  test("Gauge Chart snapshot", () => {
    createSnapshot(
      <Chart
        type="Gauge"
        chartData={[["Mock title"], [50]]}
        chartOptions={VELOCITY_OPTIONS}
        width={300}
        height={300}
      />,
    );
  });

  test("Gauge SteppedAreaChart snapshot", () => {
    createSnapshot(
      <Chart
        type="SteppedAreaChart"
        chartData={[["Mock title"], [50]]}
        chartOptions={VELOCITY_OPTIONS}
      />,
    );
  });
});
