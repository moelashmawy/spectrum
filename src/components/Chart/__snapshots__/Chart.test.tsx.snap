// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart Gauge Chart snapshot 1`] = `
<div
  class="sc-bdfCDU duXySc"
/>
`;

exports[`Chart Gauge SteppedAreaChart snapshot 1`] = `
<div
  class="sc-bdfCDU duXySc"
/>
`;
